src/Pelagos/Util/Search.php

```diff
@@ -235,13 +265,85 @@ private function getFundingOrgInfo(array $aggregations): array
                 'count' => $aggregations[$fundingOrg->getId()]
             );
         }
-
         //Sorting based on highest count
         array_multisort(array_column($fundingOrgInfo, 'count'), SORT_DESC, $fundingOrgInfo);
 
         return $fundingOrgInfo;
     }
 
+    /**
+     * Get the aggregations for the query.
+     *
+     * @param Query $query The query built based on the search terms and parameters.
+     *
+     * @return array
+     */
+    public function getStatusAggregations(Query $query): array
+    {
+        $userPaginator = $this->getPaginator($query);
+
+        $statusBucket = array_column(
+            $this->findKey($userPaginator->getAdapter()->getAggregations(), 'status')['buckets'],
+            'doc_count',
+            'key'
+        );
+
+        return $this->getStatusInfo($statusBucket);
+    }
+
+    /**
+     * Get dataset availability status information for the aggregations.
+     *
+     * @param array $aggregations Aggregations for each availability status.
+     *
+     * @return array
+     */
+    private function getStatusInfo(array $aggregations): array
+    {
+        $datasetCount = function ($status) use ($aggregations) {
+            if (array_key_exists($status, $aggregations)) {
+                return $aggregations[$status];
+            }
+        };
+
+        $statusInfo = [
+            [
+                'id' => 1,
+                'name' => 'Unavailable & Expected Soon',
+                'count' => $datasetCount(DatasetSubmission::AVAILABILITY_STATUS_NOT_AVAILABLE)
```

Comment:
This count sometimes return null, and not zero. That looks odd on the search page.

Reply:
Good catch, returning 0 in the else case now 👍